pkg/api/validation/validation.go

```diff
@@ -51,3 +52,29 @@ func validateSecret(secret *corev1.Secret, fldPath *field.Path) field.ErrorList
 
 	return allErrs
 }
+
+func validateOnmetalMachineClassSpec(spec *v1alpha1.ProviderSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if spec.RootDisk.VolumeClassName == "" {
```

Comment:
As `spec.RootDisk` can be `nil` this check might cause a nil dereference.

Reply:
Validating spec.RootDisk != nil before checking VolumeClassName